packages/kbn-handlebars/src/types.ts

```diff
@@ -5,66 +5,49 @@
+}
+
 /**
  * A custom version of the Handlesbars module with an extra `compileAST` function and fixed typings.
  */
 declare module 'handlebars' {
+  /**
+   * Compiles the given Handlebars template without the use of `eval`.
+   *
+   * @returns A render function with the same API as the return value from the regular Handlebars `compile` function.
+   */
   export function compileAST(
     input: string | hbs.AST.Program,
-    options?: ExtendedCompileOptions
-  ): (context?: any, options?: ExtendedRuntimeOptions) => string;
+    options?: CompileOptions
+  ): TemplateDelegateFixed;
 
   // --------------------------------------------------------
-  // Override/Extend inherited types below that are incorrect
+  // Override/Extend inherited funcions and interfaces below that are incorrect.
+  //
+  // Any exported `const` or `type` types can't be overwritten, so we'll just
+  // have to live with those and cast them to the correct types in our code.
+  // Some of these fixed types, we'll instead export outside the scope of this
+  // 'handlebars' module so consumers of @kbn/handlebars at least have a way to
+  // access the correct types.
   // --------------------------------------------------------
 
-  export interface TemplateDelegate<T = any> {
-    (context?: T, options?: RuntimeOptions): string; // Override to ensure `context` is optional
-    blockParams?: number; // TODO: Can this really be optional?
-    partials?: any; // TODO: Narrow type to something better than any?
-  }
-
-  export interface HelperOptions {
-    name: string;
-    loc: { start: hbs.AST.SourceLocation['start']; end: hbs.AST.SourceLocation['end'] };
-    lookupProperty: LookupProperty;
-  }
-
-  export interface HelperDelegate {
-    // eslint-disable-next-line @typescript-eslint/prefer-function-type
-    (...params: any[]): any;
-  }
-
-  export function registerPartial(spec: { [name: string]: Handlebars.Template }): void; // Ensure `spec` object values can be strings
-}
-
-export type NodeType = typeof kHelper | typeof kAmbiguous | typeof kSimple;
-
-type LookupProperty = <T = any>(parent: { [name: string]: any }, propertyName: string) => T;
-
-export type ProcessableStatementNode =
-  | hbs.AST.MustacheStatement
-  | hbs.AST.PartialStatement
-  | hbs.AST.SubExpression;
-export type ProcessableBlockStatementNode = hbs.AST.BlockStatement | hbs.AST.PartialBlockStatement;
-export type ProcessableNode = ProcessableStatementNode | ProcessableBlockStatementNode;
-export type ProcessableNodeWithPathParts = ProcessableNode & { path: hbs.AST.PathExpression };
-export type ProcessableNodeWithPathPartsOrLiteral = ProcessableNode & {
-  path: hbs.AST.PathExpression | hbs.AST.Literal;
-};
-
-export interface Helper {
-  fn?: Handlebars.HelperDelegate;
-  context: any[];
-  params: any[];
-  options: AmbiguousHelperOptions;
-}
+  export interface HelperDelegate extends HelperDelegateFixed {} // eslint-disable-line @typescript-eslint/no-empty-interface
```

Comment:
Would it be a good idea to use tsdocs style commenting on the exported types?

Reply:
Good idea... done 👍